src/kv/LevelDBStore.cc

```diff
@@ -173,6 +182,18 @@ void LevelDBStore::close()
     cct->get_perfcounters_collection()->remove(logger);
 }
 
+int LevelDBStore::repair()
+{
+  leveldb::Options ldoptions;
+  int r = load_leveldb_options(false, ldoptions);
+  if (r) {
+    dout(1) << "load leveldb options failed" << dendl;
+    return r;
+  }
+  leveldb::Status status = leveldb::RepairDB(path, ldoptions);
+  return status.ok() ? 0 : 1;
```

Comment:
nit, might want to translate `code()` to human/machine readable errno code, or pass an `ostream*` to `repair()` to we can return `status.ToString()` if anything goes wrong. please note, the caller site does not understand what `1` stands for. we take `1` as `EPERM` in general.

Reply:
thank you, Kefu. add a `ostream&` parameter as `KeyValueStore::open`